pkg/ingester/circuitbreaker.go

```diff
@@ -26,44 +26,49 @@ const (
 	circuitBreakerResultOpen         = "circuit_breaker_open"
 	circuitBreakerDefaultPushTimeout = 2 * time.Second
 	circuitBreakerDefaultReadTimeout = 30 * time.Second
+	circuitBreakerPushRequestType    = "push"
+	circuitBreakerReadRequestType    = "read"
 )
 
 type circuitBreakerMetrics struct {
 	circuitBreakerTransitions *prometheus.CounterVec
 	circuitBreakerResults     *prometheus.CounterVec
 }
 
-func newCircuitBreakerMetrics(r prometheus.Registerer, currentStateFn func() circuitbreaker.State) *circuitBreakerMetrics {
+func newCircuitBreakerMetrics(r prometheus.Registerer, currentStateFn func(string) circuitbreaker.State, requestTypes []string) *circuitBreakerMetrics {
 	cbMetrics := &circuitBreakerMetrics{
 		circuitBreakerTransitions: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
 			Name: "cortex_ingester_circuit_breaker_transitions_total",
 			Help: "Number of times the circuit breaker has entered a state.",
-		}, []string{"state"}),
+		}, []string{"request_type", "state"}),
 		circuitBreakerResults: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
 			Name: "cortex_ingester_circuit_breaker_results_total",
 			Help: "Results of executing requests via the circuit breaker.",
-		}, []string{"result"}),
+		}, []string{"request_type", "result"}),
 	}
-	circuitBreakerCurrentStateGaugeFn := func(state circuitbreaker.State) prometheus.GaugeFunc {
+	circuitBreakerCurrentStateGaugeFn := func(requestType string, state circuitbreaker.State) prometheus.GaugeFunc {
```

Comment:
```suggestion
	circuitBreakerCurrentStateGauge := func(requestType string, state circuitbreaker.State) prometheus.GaugeFunc {
```
or
```suggestion
	circuitBreakerCurrentStateGaugeFunc := func(requestType string, state circuitbreaker.State) prometheus.GaugeFunc {
```
If we want to be consistent with the type name.